code/cycler/src/wattrex_battery_cycler/mid/mid_str/mid_str_facade.py

```diff
@@ -0,0 +1,348 @@
+            CyclerDataExtMeasC, CyclerDataAllStatusC, CyclerDataExpStatusE, CyclerDataProfileC,
+            CyclerDataBatteryC, CyclerDataDeviceC, CyclerDataDeviceTypeE, CyclerDataExperimentC,
+            CyclerDataCyclerStationC, CyclerDataInstructionC, CyclerDataPwrRangeC,
+            CyclerDataPwrModeE, CyclerDataPwrLimitE, CyclerDataLinkConfC)
+
+#######################          MODULE IMPORTS          #######################
+from .mid_str_mapping import (mapping_alarm, mapping_status, mapping_gen_meas, mapping_instr_db,
+        mapping_cs_db, mapping_dev_db, mapping_batt_db, mapping_experiment, mapping_instr_modes,
+        mapping_instr_limit_modes)
+
+#######################              ENUMS               #######################
+
+#######################             CLASSES              #######################
+
+class MidStrDbElementNotFoundErrorC(Exception):
+    """Exception raised for errors when an input not found in the database.
+
+    Attributes:
+        message -- explanation of the error
+    """
+    def __init__(self, message):
+        super().__init__(message)
+
+class MidStrFacadeC:
+    '''
+    This class is used to interface with the database.
+    '''
+    def __init__(self, cycler_station_id: int,
+                 master_file : str = ".cred.yaml", cache_file : str = ".cred.yaml") -> None:
+        log.info("Initializing DB Connection...")
+        self.cs_id = cycler_station_id
+        self.__master_db: DrvDbSqlEngineC = DrvDbSqlEngineC(db_type=DrvDbTypeE.MASTER_DB,
+                                                            config_file= master_file)
+        self.__cache_db: DrvDbSqlEngineC = DrvDbSqlEngineC(db_type=DrvDbTypeE.CACHE_DB,
+                                                            config_file= cache_file)
+        self.all_status: CyclerDataAllStatusC = CyclerDataAllStatusC()
+        self.gen_meas: CyclerDataGenMeasC = CyclerDataGenMeasC()
+        self.ext_meas: CyclerDataExtMeasC = CyclerDataExtMeasC()
+        self.meas_id: int = 0
+
+    def get_start_queued_exp(self) -> Tuple[CyclerDataExperimentC, CyclerDataBatteryC,
+                                            CyclerDataProfileC]:
+        '''
+        Get the oldest queued experiment, assigned to the cycler station where this 
+        cycler would be running, and change its status to RUNNING in database.
+        '''
+        self.meas_id = 0
+        exp = None
+        battery = None
+        profile = None
+        try:
+            stmt =  select(DrvDbMasterExperimentC)\
+                        .where(DrvDbMasterExperimentC.Status == DrvDbExpStatusE.QUEUED.value)\
+                        .where(DrvDbMasterExperimentC.CSID == self.cs_id)\
+                        .order_by(DrvDbMasterExperimentC.DateCreation.asc())
+            exp_result = self.__master_db.session.execute(stmt).first()[0]
+            if exp_result is None:
+                raise MidStrDbElementNotFoundErrorC(("No experiment found for cycler station "
```

Comment:
Raise exception, but catch it on str_node because is an normal behviour. Response to APP with empty experiment or something like that

Reply:
Solved, if no experiment send none to app